services/webhook/deliver.go

```diff
@@ -288,19 +294,79 @@ func webhookProxy() func(req *http.Request) (*url.URL, error) {
 	}
 }
 
+func isWebhookRequestAllowed(allowedHostList []string, allowedHostIPNets []*net.IPNet, host string, ip net.IP) bool {
+	var allowed bool
+	ipStr := ip.String()
+loop:
+	for _, allowedHost := range allowedHostList {
+		switch allowedHost {
+		case "":
+			continue
+		case "all":
+			allowed = true
+			break loop
+		case "global":
+			if allowed = ip.IsGlobalUnicast() && !util.IsIPPrivate(ip); allowed {
+				break loop
+			}
+		case "private":
+			if allowed = util.IsIPPrivate(ip); allowed {
+				break loop
+			}
+		case "loopback":
+			if allowed = ip.IsLoopback(); allowed {
+				break loop
+			}
+		default:
+			if ok, _ := filepath.Match(allowedHost, host); ok {
+				allowed = true
+				break loop
+			}
+			if ok, _ := filepath.Match(allowedHost, ipStr); ok {
+				allowed = true
+				break loop
+			}
```

Comment:
these two blocks could potentially be written as three lines each by replacing `ok` with `allowed`.
What's your opinion on this?